src/iceberg/arrow_c_data.h

```diff
@@ -66,6 +71,47 @@ struct ArrowArray {
   void* private_data;
 };
 
-}  // extern "C"
-
 #endif  // ARROW_C_DATA_INTERFACE
+
+#ifndef ARROW_C_STREAM_INTERFACE
+#  define ARROW_C_STREAM_INTERFACE
+
+struct ArrowArrayStream {
```

Comment:
Should we mention [0] in the \file description?

I have a feeling that this is not related to PR $titile, so maybe another PR for this?

[0] https://arrow.apache.org/docs/format/CStreamInterface.html

Reply:
I have to add this, otherwise compiling files with headers from `nanoarrow` will complain missing definition of `ArrowArrayStream`.

Reply:
OK, then the patch LGTM.

Reply:
I have removed ArrowArrayStream. I think internally we can just include `nanoarrow.h` so we don't have to add ArrowArrayStream and confuse the downstream.